core/src/test/scala/unit/kafka/server/AlterReplicaLogDirsRequestTest.scala

```diff
@@ -36,6 +37,13 @@ class AlterReplicaLogDirsRequestTest extends BaseRequestTest {
 
   val topic = "topic"
 
+  private def findErrorForPartition(response: AlterReplicaLogDirsResponse, tp: TopicPartition): Errors = {
+    Errors.forCode(response.data.results.asScala.groupBy(x => x.topicName).get(tp.topic).get
```

Comment:
I find the implementation that you did in `AuthorizerIntegrationTest` at L195 easier to read. Could we use it here as well to stay consistent?